os/Makefile.unix

```diff
@@ -533,7 +533,7 @@ ota_img:
 # pass1dep: Create pass1 build dependencies
 # pass2dep: Create pass2 build dependencies
 
-pass1dep: context tools/mkdeps$(HOSTEXEEXT)
+pass1dep: context tools/mkdeps$(HOSTEXEEXT) compression_tool
```

Comment:
We can call `make pass2`, then compression is not working. Right?

Reply:
make pass2, neither is compression tool built and neither are elf app binaries built, which call compression tool.

make pass1 will first build the compression tool and then elf app binaries which will call the compression tool.

Reply:
That's why I mention. I think it does not make sense.
Even if I want to get pass2 binary only, because of this, I should call pass1 together. Then there is no reason to split them.
How about move this at `context` or other target?
`context`'s main purpose is preparing build by linking, copying or something else.

Reply:
or add that at both pass1dep and pass2dep

Reply:
@sunghan-chang 
Regarding "Even if I want to get pass2 binary only, because of this, I should call pass1 together"

I don't think this is the case.

make pass2 will only build os files; it will not build elf binaries under apps/ which need the compression tool.

make pass1 will build apps/ which will make elf binaries which need compression tool.

So, I think compression tool should be added for pass1dep target only, not pass2dep

Reply:
Ok. I understood what you say. But we are checking loading kernel binary instead of XIP. Then, it is needed for both. @kishore-sn is working on it. When we finish to make kernel loading, we can add it on pass2 as well.

Reply:
Ok...  understood